nim_status/lib/contacts.nim

```diff
@@ -0,0 +1,129 @@
+    Timestamp = "timestamp",
+    FCMToken = "fcmToken"
+
+  ContactDeviceInfo* = object
+    # The installation id of the device
+    installationId* {.serializedFieldName($ContactDeviceInfoType.InstallationId).}: string
+    # Represents the last time we received this info
+    timestamp* {.serializedFieldName($ContactDeviceInfoType.Timestamp).}: int64
+    # to be used for push notifications
+    fcmToken* {.serializedFieldName($ContactDeviceInfoType.FCMToken).}: string
+
+  # Contact has information about a "Contact". A contact is not necessarily one
+  # that we added or added us, that's based on SystemTags.
+  Contact* = object
+    # ID of the contact. It's a hex-encoded public key (prefixed with 0x).
+    id* {.serializedFieldName($ContactType.Id), dbColumnName($ContactsCol.Id).}: string 
+    # Ethereum address of the contact
+    address* {.serializedFieldName($ContactType.Address), dbColumnName($ContactsCol.Address).}: Option[Address]
+    # ENS name of contact
+    name* {.serializedFieldName($ContactType.Name), dbColumnName($ContactsCol.Name).}: Option[string]
+    # Whether we verified the name of the contact
+    ensVerified* {.serializedFieldName($ContactType.EnsVerified), dbColumnName($ContactsCol.EnsVerified).}: bool
+    # the time we last verified the name
+    ensVerifiedAt* {.serializedFieldName($ContactType.EnsVerifiedAt), dbColumnName($ContactsCol.EnsVerifiedAt).}: int64
+    # last clock value of when we received an ENS name for the user
+    lastEnsClockValue* {.serializedFieldName($ContactType.LastEnsClockValue).}: int64
+    # how many times we retried the ENS
+    ensVerificationRetries* {.serializedFieldName($ContactType.EnsVerificationRetries).}: int64
+    # Generated username name of the contact
+    alias* {.serializedFieldName($ContactType.Alias), dbColumnName($ContactsCol.Alias).}: Option[string]
+    # Identicon generated from public key
+    identicon* {.serializedFieldName($ContactType.Identicon), dbColumnName($ContactsCol.Identicon).}: string 
+    # base64 encoded photo
+    photo* {.serializedFieldName($ContactType.Photo), dbColumnName($ContactsCol.Photo).}: Option[string]
+    # last time we received an update from the contact. Updates should be discarded if last updated is less than the one stored
+    lastUpdated* {.serializedFieldName($ContactType.LastUpdated), dbColumnName($ContactsCol.LastUpdated).}: int64
+    # contains information about whether we blocked/added/have been added
+    systemTags* {.serializedFieldName($ContactType.SystemTags), dbColumnName($ContactsCol.SystemTags).}: seq[string]
+    deviceInfo* {.serializedFieldName($ContactType.DeviceInfo), dbColumnName($ContactsCol.DeviceInfo).}: seq[ContactDeviceInfo]
+    tributeToTalk* {.serializedFieldName($ContactType.TributeToTalk), dbColumnName($ContactsCol.TributeToTalk).}: Option[string]
+    localNickname* {.serializedFieldName($ContactType.TributeToTalk), dbColumnName($ContactsCol.TributeToTalk).}: Option[string]
```

Comment:
Is e.g. `ContactType.TributeToTalk` correct for `localNickname`?

Reply:
No. `localNickname` is a `string`